### ISSUE TITLE
Tag a Document

### ISSUE THREAD

**Issue by pithub** [the-sett/elm-pretty-printer#7]
Hi @rupertlssmith, for my port of the Elm compiler from Haskell to Elm I need a function like
```elm
setTag : t -> Doc t -> Doc t
```
which sets the tag for all the strings in an already assembled document.

If you like, I could create a pull request.

Instead of or additionally to this function I could implement a function like
```elm
updateTag : (Maybe t -> Maybe t) -> Doc t -> Doc t
```
or even
```elm
updateTag : (String -> Maybe t -> Maybe t) -> Doc t -> Doc t
```
which would give more control over the tag replacement, but I'm not sure whether something like this would be needed.

What do you think?

**Comment by rupertlssmith** [comment]
I think this would be useful. Might as well implement this one for the sake of exposing the most useful function we can:

updateTag : (String -> Maybe t -> Maybe t) -> Doc t -> Doc t

If you create a PR, I will merge it in and make a new release with it available, thanks.

**Comment by rupertlssmith** [comment]
Fixed by:

https://github.com/the-sett/elm-pretty-printer/pull/8

Published as `the-sett/elm-pretty-printer` 3.1.0

**Comment by pithub** [comment]
Wow, that was fast, Rupert. As if you didn't have more important work to do 😅 Thank you very much 👍
